### ISSUE TITLE
Review

### ISSUE THREAD

**Issue by tcrasset** [RezaNasrollahi/Wine_Quality_NN#1]
# Structured data review

## Good

- Very nice EDA. Although you could have put it into a separate document.
- Very nice plots when changing learning rate
- The learning rate decreases, nice!
- Like the fact that you tried hyper-parameter tuning. If you want to automate it, you can use `ray-tune`, `optuna`, etc.

## To be improved

- This path does not exist

  ```python
  df=pd.read_csv("../Wine Quality/wine.csv")
  ```

- What about other hyper-parameters than the learning rate ?
- You can specify the label in the `plt.plot()` function with the `label=` argument. This makes it harder to mislabel your graphs

```python
for i in range(len(rates)):
    plt.plot(range(1, number_epoch+1), a_ts_loss[i, :])
plt.legend(['LR = 0.1', 'LR = 0.01', 'LR = 0.001'])
```

- What you call `test_loader` is actually a validation set, i.e. a set that you use while training to probe the _unknown data_. However, you should have a final dataset at the end of training to measure the final error

- Your last layer should have an activation function. As you are doing multi-class classification, it should most likely be `Softmax`

```python
        out = self.layer_3(x2)
```
